### PR TITLE
Backpack block

### PR DESCRIPTION
**Pull request by Tarmunora**
[ENHANCEMENT]
## About the PR 
Adds special functionality to blocking with backpacks (and children thereof, i.e.: satchels), that gives an extra point of block for every 3 items in the bag, at the cost of an item being randomly ejected from the bag when you are struck.
PR also cleans up the rock-it launcher a bit, so it no longer has a buggy HUD (ran into similar issues when ejecting items from bag, so fixed in both spots)

Devnote: Adds signal `COMSIG_MOB_ATTACKED_PRE`, sent when an attack on mob has hit, but before armor/etc is calculated.

## Why's this needed? 
Idea from [forum thread](https://forum.ss13.co/showthread.php?tid=14372), thought that it would be cool and decently balanced

## Changelog

```
(u)Tarmunora:
(+)Added special block functionality to backpacks/satchels.
```

### CODE REVIEW

File: _stdlib/_setup.dm [hunk 1]
```diff
@@ -765,7 +765,7 @@
 #define DAMAGE_BURN 8					// a) this is an excellent idea and b) why do we still use damtype strings then
 #define DAMAGE_CRUSH 16					// crushing damage is technically blunt damage, but it causes bleeding
 #define DEFAULT_BLOOD_COLOR "#990000"	// speak for yourself, as a shapeshifting illuminati lizard, my blood is somewhere between lime and leaf green
-
+#define DAMAGE_TYPE_TO_STRING(x) (x == 1 ? "blunt" : x == 2 ? "cut" : x == 4 ? "stab" : x == 8 ? "burn" : x == 16 ? "crush" : "")
```
Comment (ZeWaka) [inline _stdlib/_setup.dm:768]: Why are you using the numbers here and not the defines?

File: code/obj/item/storage/small_storage_parent.dm [hunk 2]
```diff
@@ -248,9 +248,10 @@
 		attack_hand(user)
 
 	proc/get_contents()
+		RETURN_TYPE(/list)
```
Comment (ZeWaka) [inline code/obj/item/storage/small_storage_parent.dm:251]: nice

**Comment by Tarmunora** [comment]
Uh that's a *good question*. 




On Fri, Apr 24, 2020 at 4:19 AM -0500, "ZeWaka" <notifications@github.com> wrote:















@ZeWaka commented on this pull request.





In _stdlib/_setup.dm:
> @@ -765,7 +765,7 @@
 #define DAMAGE_BURN 8					// a) this is an excellent idea and b) why do we still use damtype strings then
 #define DAMAGE_CRUSH 16					// crushing damage is technically blunt damage, but it causes bleeding
 #define DEFAULT_BLOOD_COLOR "#990000"	// speak for yourself, as a shapeshifting illuminati lizard, my blood is somewhere between lime and leaf green
-
+#define DAMAGE_TYPE_TO_STRING(x) (x == 1 ? "blunt" : x == 2 ? "cut" : x == 4 ? "stab" : x == 8 ? "burn" : x == 16 ? "crush" : "")



Why are you using the numbers here and not the defines?





In code/obj/item/storage/small_storage_parent.dm:
> @@ -248,9 +248,10 @@
 		attack_hand(user)
 
 	proc/get_contents()
+		RETURN_TYPE(/list)



nice

File: _stdlib/_setup.dm (hunk 1 above)
Comment (Tarmunora) [reply _stdlib/_setup.dm]: oop

File: code/obj/item/storage/small_storage_parent.dm (hunk 2 above)
Comment (Tarmunora) [reply code/obj/item/storage/small_storage_parent.dm:251]: spacemanDMM was yelling at me

**Pull request merged**